### ISSUE TITLE
`==` not implemented for any (abstract) types

### ISSUE THREAD

**Issue by leethargo** [JuliaGeo/GeoInterface.jl#21]
I was just surprised by
```julia
julia> [0.0, 0.0] == [0.0, 0.0]
true

julia> Point([0.0, 0.0]) == Point([0.0, 0.0])
false
```

Turns out that `==` is not implemented for any of the types in GeoInterface, so Julia applies the `===` fallback. Is this intentional?

I think a value-based comparison could be had quite easily:
```julia
julia> import Base: ==

julia> ==(x::AbstractGeometry, y::AbstractGeometry) = coordinates(x) == coordinates(y)
== (generic function with 155 methods)

julia> Point([0.0, 0.0]) == Point([0.0, 0.0])
true
```

This would be useful for me, as I try to use `Point`s as keys to `Dict`s, but I don't want to commit type piracy. I can open PR if this change is welcome.

**Comment by yeesian** [comment]
> I think a value-based comparison could be had quite easily

That's a good question. It is more tricky when we look beyond points (since their representations won't be unique), but I think it's fine for points.

I'll be happy to review a PR for it. While you're at it -- [should it be done for approx too?](https://github.com/Toblerity/Shapely/pull/537/files)

**Comment by leethargo** [comment]
OK, I did not think as deeply about it yet. For higher-level geometries, such as polygons, I think a comparison of the raw coordinates is still valuable. It would be an equality of representation, rather than an equivalence, but still. If the objects come from the same source, they will also compare as equal.

If we did inclusion testing, then the complexity (in time and algorithm) increases a lot, so I would rather use a separate function for that (and maybe a separate package).

Adding `approx` is an excellent idea.
